### PR TITLE
Fix self-runner

### PR DESCRIPTION
**Pull request by scarrazza**
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by andrea-pasquale — APPROVED** [review]

**Comment by scarrazza** [comment]
@rodolfocarobene are those features related to the fpga implementation?

**Comment by rodolfocarobene** [comment]
6 tests on qpu are failing, 2 of them are related to rfsoc driver. They fail correctly and I need to fix them, but the problem is that GitHub has marked the tests on qpu succesful... Not sure why

**Comment by scarrazza** [comment]
Oh, I see, in fact we have never implemented in the self-runner the mechanism which shows a failed test on github if pytest fails, but instead we simply post the report for human inspection. 

From a technical perspective it is possible to implement, but this may require the parsing of the slurm output file, but I would suggest to merge as it is given that the number of self-runner jobs is limited so far.  (I will open an issue to implement this feature in the future).

File: .github/workflows/selfhosted.yml [hunk 1]
```diff
@@ -47,13 +47,26 @@ jobs:
...
-        python-version: 3.9
+        python-version: "3.10"
+    - name: Install and configure poetry
+      uses: snok/install-poetry@v1
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+    - name: Disable modern installer
+      run: |
+        poetry config installer.modern-installation false
```
Comment (AleCandido) [inline .github/workflows/selfhosted.yml:58]: Is there a specific reason for this? The issue with `furo` should have been solved, do we have any other bloody package?

**Review by AleCandido — APPROVED** [review]
Apart from the single comment above, the workflow is mostly fine. Thanks!

File: .github/workflows/selfhosted.yml (hunk 1 above)
Comment (scarrazza) [reply .github/workflows/selfhosted.yml:58]: No, it is just copy/paste from https://github.com/qiboteam/workflows/blob/main/.github/workflows/rules-poetry.yml#L46.

**Comment by scarrazza** [comment]
Thanks @AleCandido, I have simplified the workflow. @andrea-pasquale @rodolfocarobene I will merge this now in your runcard PR, please continue there for tests fixes.

File: .github/workflows/selfhosted.yml (hunk 1 above)
Comment (AleCandido) [reply .github/workflows/selfhosted.yml]: Ok, so we should have updated the other one first @Edoardo-Pedicillo

However, I saw you actually simplified, and the runner passed with the same amount of errors as before. So, please keep going :)

Comment (scarrazza) [reply .github/workflows/selfhosted.yml]: > However, I saw you actually simplified, and the runner passed with the same amount of errors as before. So, please keep going :)

@AleCandido if I had time that would be very fun, indeed ;-).

**Pull request merged**